### PR TITLE
Change global option change notification to batch all option changes

### PR DESCRIPTION
**Pull request by tmat**
Batching allows to perform a single update operation when multiple global options change. This will be useful when updating solution snapshot with option values affecting analyzers.

### CODE REVIEW

**Comment by tmat** [comment]
@CyrusNajmabadi ptal

File: src/LanguageServer/Protocol/Handler/InlayHint/InlayHintRefreshQueue.cs [hunk 1]
```diff
@@ -33,18 +33,19 @@ public override void Dispose()
...
+                option.Equals(InlineHintsOptionsStorage.ForLiteralParameters) ||
+                option.Equals(InlineHintsOptionsStorage.ForOtherParameters) ||
+                option.Equals(InlineHintsOptionsStorage.ForObjectCreationParameters) ||
+                option.Equals(InlineHintsOptionsStorage.SuppressForParametersThatDifferOnlyBySuffix) ||
+                option.Equals(InlineHintsOptionsStorage.SuppressForParametersThatMatchArgumentName) ||
+                option.Equals(InlineHintsOptionsStorage.SuppressForParametersThatMatchMethodIntent) ||
+                option.Equals(InlineHintsOptionsStorage.EnabledForTypes) ||
+                option.Equals(InlineHintsOptionsStorage.ForImplicitVariableTypes) ||
+                option.Equals(InlineHintsOptionsStorage.ForLambdaParameterTypes) ||
+                option.Equals(InlineHintsOptionsStorage.ForImplicitObjectCreation)))
```
Comment (CyrusNajmabadi) [inline src/LanguageServer/Protocol/Handler/InlayHint/InlayHintRefreshQueue.cs:48]: indent

File: src/VisualStudio/Core/Def/InheritanceMargin/InheritanceMarginViewMargin.cs [hunk 2]
```diff
@@ -124,8 +124,9 @@ private void OnLayoutChanged(object sender, TextViewLayoutChangedEventArgs e)
 
     private void OnGlobalOptionChanged(object sender, OptionChangedEventArgs e)
     {
-        if (e.Option.Equals(InheritanceMarginOptionsStorage.ShowInheritanceMargin) ||
-            e.Option.Equals(InheritanceMarginOptionsStorage.InheritanceMarginCombinedWithIndicatorMargin))
+        if (e.HasOption(option =>
+            option.Equals(InheritanceMarginOptionsStorage.ShowInheritanceMargin) ||
+            option.Equals(InheritanceMarginOptionsStorage.InheritanceMarginCombinedWithIndicatorMargin)))
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/InheritanceMargin/InheritanceMarginViewMargin.cs:129]: indent

File: src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerCommandHandler.cs [hunk 3]
```diff
@@ -107,16 +108,18 @@ private void UnadviseBroadcastMessages()
...
-            {
-                UnadviseBroadcastMessages();
+                if (enabled)
+                {
+                    AdviseBroadcastMessages();
+                }
+                else
+                {
+                    UnadviseBroadcastMessages();
+                }
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerCommandHandler.cs:122]: i'm wary of this loop.  shoudl we compute the value once in the loop, and then advise/unadvise once after instead?

File: src/LanguageServer/Protocol/Handler/InlayHint/InlayHintRefreshQueue.cs (hunk 1 above)
Comment (tmat) [reply src/LanguageServer/Protocol/Handler/InlayHint/InlayHintRefreshQueue.cs:48]: Indent one more tab?

File: src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerCommandHandler.cs (hunk 3 above)
Comment (tmat) [reply src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerCommandHandler.cs:122]: In practice the options are unique, but I guess that's not guaranteed anywhere and they might repeat.

File: src/LanguageServer/Protocol/Handler/InlayHint/InlayHintRefreshQueue.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/LanguageServer/Protocol/Handler/InlayHint/InlayHintRefreshQueue.cs:48]: yes please.  it helps see that it's in a lambda, as opposed to a check that is a sibling of the top check.  (i would like the indentation engine to do this automatically, but we've never gotten around to it).

File: src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerCommandHandler.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerCommandHandler.cs:122]: ok.  as long as you're ok with this going on/off multiple times, that's ok with me.  i genuinely dont' know what the semantics are ere.

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by tmat** [comment]
@dibarbet ptal at the last commit

**Review by dibarbet — APPROVED** [review]

**Pull request merged**